### PR TITLE
Timeline

### PR DESCRIPTION
**Pull request by romanroe**
**Styling** 

- [x] Horizontal lines missing in timelines legend (table header above timeline)
- [x] Bars with only one date should have the same size in all zoom stages.

**Hierarchy work package Features**

- [ ] change display of group parent when the group is folded
- [ ] highlight groups

**Relations work package Features**

- [ ] display work package relations
- [ ] GUI-CRUD work package relations ?? (drag'n'drop vs MS Project)

**Additional Features**
- [ ] Save timeline settings

- [ ] The timelines module (opened through sidebar) has different defaults for columns than the "normal" list view.
- [ ] Show additional work package attributes in the timeline (subject, status, assignee, % done ...)
- [ ] Configure/save which work package attributes are shown in the timliens view

**Bugs**

- [ ] Wrong cursor icons on Windows
- [ ] Zoom toolbar gets lost from time to time.
- [ ] When adding a relation in the split view the timeline is not updated accordingly.
- [x] Adding a date by drag and drop does not work after changing the work package type.

### CODE REVIEW

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/3606d5c558d5cf4b77610bf78fc8dcc8c1cc358e) (deployed commit: 6e74bd3a3faec7a08869f2e2869aebce690412f0).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/ae8421978aa4b2ea21dc8a7ba29d014f725eb189) (deployed commit: 8ac146e80053aad677c2a22a4b6ab41e434625ee).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/7d9a40b76656c223102e2298ae6b0b42f7b315af) (deployed commit: 5b9acfd373456fbe6df23ce17631a352804b8b55).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/2e4fc3224e17818b6ca2e66c867db1ca398ec148) (deployed commit: bbf942feb277c3d6c8ae0d0a4432d5d7c3770eff).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/9a2266836c59dc2d700029f5a74eebbfcc6b53f7) (deployed commit: 50e7ba47ddbe4491ba6823c9a50b1d4b407a1b92).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/bc22364ea84fac33b3eb5751d1a5b9eebddc5fd4) (deployed commit: f5c50607cb324cb35d61ef99c1e5e813ee783b10).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/3386f5cf7d5062b0fc46f0523514b40ef7a30bdf) (deployed commit: bab428b3de5086fcd6ccf20bc258c2b1bbb540c0).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/d1dabed0b48b1488bd9735e219e67d116d1566c0) (deployed commit: 418f637f2fca57ee0161e1e7c98cdd003e2b0e89).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/3de5afb1f95133fb573fa57b1d4764fe7eb81702) (deployed commit: 14d5dc27ca2159b35f262e112a5459e98acdf92d).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/e9b7e0bc0454dc6260ae6d32494ee1910fb3bf81) (deployed commit: 80f5b11997722763ab5b85500c3bb0f956c61f56).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/8e008afb92c4cfa97f75298707256e80781c322d) (deployed commit: 80f5b11997722763ab5b85500c3bb0f956c61f56).

**Comment by kgish** [comment]
The page you were looking for doesn't exist.

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/03160debb28b99e6dd2b1f7f4edc9d7c3e18e5cd) (deployed commit: 80f5b11997722763ab5b85500c3bb0f956c61f56).

**Comment by lindenthal** [comment]
Hi @romanroe,

when zooming sometimes the legend and the vertical lines disappear. Can you please have a look at this?

image

Best
Niels

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/e25637162bf55087e4ed63545cf89db8f20a19b4) (deployed commit: ab67ff19730ac1a49d2f0e3a7bd65b82f2f74d37).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/bdf70d84c258288c2135a234bace91f6d79c9c80) (deployed commit: 0fcddd686fa994371b4228d7f65aaec9e690b518).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/8453faa7709d7b66bb6d302ef22f5f8441126138) (deployed commit: 8e091019a57c2d5bbf1137a6dec0b5ca4f58b906).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/ce8525f7d6a93ab27dac3eb943ea66942fc83397) (deployed commit: 9e85e807b196582d1364218dfdda42cd865a4b45).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/dcf9797e3531f70406c4af127b68bf15703f22e8) (deployed commit: 38e42d5117070f23f47d8ea0a67d75375c38ad04).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/4c82278c20b9216b1a1a4a558c2f946d90e39a93) (deployed commit: f0a99ccdcd8c956190d03c49deff8302cd984d18).

**Comment by romanroe** [comment]
@oliverguenther As discussed, bug: a) smallest zoom level causes the table to display a horizontal scrollbar b) timeline h-scroll shows 100/100 instead of e.g. 70/100

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/e625e9e9890fab413d35374f7eb2f1dac67ace9d) (deployed commit: c319431bd1bbf956b4c196596367be45642254fa).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/b36385324899126c676dc54b5752bcdb7a3e3eaa) (deployed commit: e7fb7c5c8ced7eff7b29682135fae907a93e5f66).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/47c8d575980bbd2d8f92003598b1bdfc9d388968) (deployed commit: 2786bf96dfd3e3e2e9b254d695d97f770ca8aca6).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/13d1ba8f0eebb53172913709c89a90c72e8b296b) (deployed commit: fdf01d33590984a8512fc71832f82ea26ebb264e).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/bbb2b754c69b938716bbbd33db419a54879a0c0c) (deployed commit: 511d767704e9c7ebb3d7f824db4cfde490f77522).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/f54618672ef843a2f99a4e91cc35f56d913efc25) (deployed commit: e74858b9d8595fc0c77ea52868be53f3fed7715e).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/7fb2890b0fa0604098b4d351e8cd1b583e30b477) (deployed commit: 41780ae1e2c35275e3bc643d7ec9f57fa5595e77).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/eaf4e2d9be4007a2e259562d8b81072c30c169ad) (deployed commit: 9c867c0ca1a8d49a0c44775de1e8314381a1e9ef).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/69fb3f83ce6db9bcf6b1f7f71c2a45dd087b9361) (deployed commit: d54610ef26afd1e9ffaf402d641092df611b4b81).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/09ac120e97ab6eefe9f4f3cb368e4d4debca8c8a) (deployed commit: 3365224314a39473612b6141d31ba18a9679bfa9).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/b9d6af44ec2dee790fb0c207429d031f615907a3) (deployed commit: 7966ede2df0a044bbb0075330fbdc770110306f5).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/14a72647ad0dcbd292cca5f5f14147d8227ea7b8) (deployed commit: 75694dbc07066be25bfcf45b6f00472aae8f3429).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/c00bcacbc56210cf5517cc896fb401ea380d778b) (deployed commit: 98c99ce283f88f6cf4826c1210b79f4c1c0958c0).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/7e70ab0bc32e801c8be112da97675bf3a3078758) (deployed commit: 89db167d72594002b91f3f7ffa2e34aafff4d03f).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/86950eaca18e7805c69d5890f45bd6735ba1eb2c) (deployed commit: d90e813e6d5d5afb3a05500b61a4ba2675d8b501).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/3d4a594226582ee1f5137cbcf7cd7ea2c13f539f) (deployed commit: dde6b9f1a1698eff4928b7040e16e229ad42a05e).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/57b23eeed95371c9d9e203570d3dc4fdd61f6148) (deployed commit: dde6b9f1a1698eff4928b7040e16e229ad42a05e).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/125352f42a7667bc6e7f9a1776ca172e219e1ab4) (deployed commit: da64e02ef8cedb58a080b15941c86d8c8b253a0f).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/26bfc5019f1b34372f95deef8f20d1a6bc19d3ed) (deployed commit: e8e798785d61ceecd585edc70d4263518e9cc852).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/fc90fc71184b5694f470f2a354a86e92ee0bdb3d) (deployed commit: 4d78390575105b5b5cdc8b60c7dd285ec7e4610b).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/297aab17283846bb879ff8d68e8dab1a845eac1f) (deployed commit: 4b1dbcfe7ead5efadf7ad0303d1d5aef5ee0a5c9).

File: frontend/app/components/wp-table/timeline/wp-timeline-global.directive.ts [hunk 1]
```diff
@@ -98,8 +98,12 @@ export class WpTimelineGlobalService {
 
       // start
       const startCell = this.cells[e.from];
-      let lastX = startCell.getRightmostPosition();
 
+      if (!startCell) {
```
Comment (oliverguenther) [inline frontend/app/components/wp-table/timeline/wp-timeline-global.directive.ts]: /cc @romanroe This caused trouble on the timeline branch, so for sake of testing the merge with hierarchies, i've added this simple check. Please revise

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/de584b865a16d383fbdaaab4856bad44afbc85db) (deployed commit: 667a674cd73c5feb1d1452d68744ec35074d1c72).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/af603c84aa774a048d225ef47f19ad0655376398) (deployed commit: f646eb2d0f8dffbcf0bb75a595e943b85f6ef987).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/63658911571f868cae6732e1072bef69de7f7f70) (deployed commit: cc11183b5438d7eee2b74bc21463288f4274e823).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/563120fb3f1684df755a3bb2fdac8ac91888903d) (deployed commit: c3c7e17aeb77522ab905de2959d1f302d57d7d29).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/ecd689badb30d7df80360dfc032a27605af8be9e) (deployed commit: c51e3d61209082fab25ca1ae2dcfbb3eae16d2d3).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/62c6f6ce77c1d7c8921d86a7f0d91d16bf6f9848) (deployed commit: 0f49e516f5de9bccd420e0152572a10594f1aaba).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/79f00c7a52d2b111d6ff0ae7fba56f7d381d2f0a) (deployed commit: a5b7592cbc74b9569f892d4cd760829240c01b9c).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/775c6db4e6c36b5c9a048f2684bfdac9f4b26757) (deployed commit: 45ca9ffea36808af1bb3ffe3b6eccdf4c0597098).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/b6d40c4e3ac61eb03ef6a047733b3803d56eaaf6) (deployed commit: b5132320bf0f86408ac66fa20ba71f9d24a3d176).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/c130d0d5b6ee86b7ca4ead8d68d4a061c3fa81fc) (deployed commit: 1472c49583512b117854a69ad0c6390154bbcf7b).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/6bc29aee14f08bafbf28ac060b9410f0d70032f8) (deployed commit: d569e7704c8d311727bc90d4e134725df3a4ba8c).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/37fe31e8eb70c39a0b5c3a7b4dea6522a45c9e2f) (deployed commit: 8c3fcee17aac52d338e006860b4bde0858983e48).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/0285c32f2387d397a3f111a96baed864aba19776) (deployed commit: 4b0cd9a8373a3fef36024176e664890ade0589ce).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/541a24c5f3ba91b818c2126217797f878d2ceaeb) (deployed commit: 37952d6be5d92205f325ca085382f0804211d5dd).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/acff130e9dc1b03f50fe44006d3a268afda008a0) (deployed commit: a6e02235d78101bbd1a8cf5ac4879c4d1c9318fb).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/3395febc52e4e59fe880cf557c253f3757b6f147) (deployed commit: 0259d09e9a5a96714ddfd940910625dcff9d7520).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/2a6e91fa398910524a963d6ba5d59ca9304579c3) (deployed commit: 810dad7d7ead247e034f77c3f273f0522c6ebde5).

**Comment by oliverguenther** [comment]
Finished the merge after the merge of sticky headers in https://github.com/opf/openproject/pull/5213

This required some repositioning (mostly z-index) on the thead and header.

**Still in progress**: Since the interactive-table is no longer in use, we need another mechanism to detect height changes (/cc @romanroe).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/a2b6b6ffdbe89aea54ef7f435ac9277fa8ee39b1) (deployed commit: a7c5a559fd2ab349b27a2aa963de7f16796ebff0).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/dd37b132e9f094808bd5a6af2152a376f13b7841) (deployed commit: 64fc0f04ae5d55cefea11683002b752565d3d621).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/20595cc5985c5eb5e965e182c2edc7d9f3ba89a7) (deployed commit: 50e9da5ad3606a118ea6260c95471f6bf12d2943).

**Comment by crohr** [comment]
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/b253b75698545b8a3ada7b271d5b2f23381078af) (deployed commit: 847c7b34b6fbfc48de486011b7c1bd2c5c0b556c).

**Pull request merged**